packages/sdk/src/across/lpFeeCalculator.ts

```diff
@@ -14,6 +14,11 @@ export default class LpFeeCalculator {
     this.blockFinder = new BlockFinder<Block>(provider.getBlock.bind(provider));
   }
   async getLpFeePct(tokenAddress: string, bridgePoolAddress: string, amount: BigNumberish, timestamp?: number) {
+    // TODO: Accept another param, rateModelStoreAddress, construct a RateModelStore contract instance, query all
```

Comment:
see here @daywiss

Reply:
i dont understand why we need to query all events, wasnt there a function call to get the latest rate model?

Reply:
> wasnt there a function call to get the latest rate model?
The `getRateModelForBlockNumber` expects an array of `RateModelEvents` to be passed to it to find the correct one for the block number. The reason why events aren't queried in that function is because the FE and the bots would query events differently using `ethers` and `web3` contract instances respectively. Does that make sense?